ctru-rs/src/services/cam.rs

```diff
@@ -381,115 +604,111 @@ pub trait Camera {
+        match self.trimming() {
+            Trimming::Centered(trim_size) => trim_size.into(),
+            Trimming::Off => self.view_size().into(),
         }
     }
 
-    /// Returns the [`TrimmingParams`] currently set.
-    #[doc(alias = "CAMU_GetTrimmingParams")]
-    fn trimming_params(&self) -> crate::Result<TrimmingParams> {
-        unsafe {
-            let mut x_start = 0;
-            let mut y_start = 0;
-            let mut x_end = 0;
-            let mut y_end = 0;
-            ResultCode(ctru_sys::CAMU_GetTrimmingParams(
-                &mut x_start,
-                &mut y_start,
-                &mut x_end,
-                &mut y_end,
-                self.port_as_raw(),
-            ))?;
-
-            Ok(TrimmingParams {
-                x_start,
-                y_start,
-                x_end,
-                y_end,
-            })
-        }
+    /// Returns the [`Trimming`] configuration currently set.
+    fn trimming(&self) -> Trimming {
+        self.configuration().trimming
     }
 
-    /// Set the trimming bounds relatively to the center of the image.
+    /// Set trimming bounds to trim the camera photo.
     ///
     /// # Notes
     ///
-    /// The new width will be `trim_width / 2` to the left and right of the center.
-    /// The new height will be `trim_height / 2` above and below the center.
-    // TODO: This function doesn't use `TrimmingParams`. It'd be better to merge it with `set_trimming_params()` and change the `TrimmingParams` representation.
-    #[doc(alias = "CAMU_SetTrimmingParamsCenter")]
-    fn set_trimming_params_center(
-        &mut self,
-        trim_width: i16,
-        trim_height: i16,
-        cam_width: i16,
-        cam_height: i16,
-    ) -> crate::Result<()> {
-        unsafe {
-            ResultCode(ctru_sys::CAMU_SetTrimmingParamsCenter(
-                self.port_as_raw(),
-                trim_width,
-                trim_height,
-                cam_width,
-                cam_height,
-            ))?;
-            Ok(())
+    /// Trimming the image view directly on the cameras can only work if the final image has the size of one of the supported image formats.
+    /// As such, even after trimming the image (which will result in a "zoomed in" view of the original image)
+    /// the resulting picture must still be of [`ViewSize`] dimensions.
+    #[doc(alias = "CAMU_SetTrimming")]
+    fn set_trimming(&mut self, trimming: Trimming) -> crate::Result<()> {
+        // Run checks for all trimming possibilities.
+        match trimming {
+            Trimming::Centered(trim_size) => unsafe {
+                let view_size: (i16, i16) = self.view_size().into();
+                let trim_size: (i16, i16) = trim_size.into();
+
+                // Trim sizes are within the view.
+                assert!(
+                    trim_size.0 <= view_size.0 && trim_size.1 <= view_size.1,
+                    "trimmed view is bigger than the camera view"
```

Comment:
Is this something that is possible for the user to configure? Does it make sense to return an `Err` for them to handle it in that case?

Reply:
It wouldn't make sense for the user to pass a bigger trim size than the original view. The trimming is meant to cut *down* the view so to get a particular detail of the image.

Other than panicking, we could either return an `Error` or just "clamp" the trim view to the original size (e.g. if the original width is 400 and the user trims to 500, we just return the image with the original width of  400).

Maybe the second approach would be preferable?

Reply:
I've come to the conclusion that this situation cannot be handled with clamping or other measures, as the final image resolution isn't guaranteed to adhere to the imposed standards. It's best to place the responsibility on the developer by making this issue known.